src/main/java/bookbob/Main.java

```diff
@@ -67,6 +71,18 @@ public static void main(String[] args) throws IOException {
                         logger.log(Level.INFO, "NRIC of the patient is not provided");
                         break;
                     }
+
+                    String visitDateString = input.substring(visitStart + 2).trim();
+                    try {
+                        // Attempt to parse using a standard formatter
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+                        LocalDateTime time = LocalDateTime.parse(visitDateString, formatter);
+                    } catch (DateTimeParseException e) {
+                        System.out.println("Invalid visit date format. Please use 'dd-MM-yyyy HH:mm' format.");
+                        logger.log(Level.INFO, "Invalid visit date format provided");
+                        break;
```

Comment:
I like how you catch a DateTimeParseException when the user provides an invalid visit date and time format. The printed message is also helpful in guiding the user about the correct format to write.

Reply:
Thank you!